app/src/main/java/sdumchykov/androidApp/presentation/viewPager/contacts/MyContactsFragment.kt

```diff
@@ -0,0 +1,302 @@
+                    enableMultiselectViewState()
+                    parentViewModel.selectedEvent.value = true
+                    disableContactSwipe()
+                    initRecyclerView()
+                    imageViewMyContactsCancelSetOnClickListener()
+                }
+
+                override fun onContactSelectedStateChanged() {
+                    if (usersAdapter.areAllItemsUnselected()) {
+                        disableMultiselectViewState()
+                        parentViewModel.selectedEvent.value = false
+                        enableContactSwipe()
+                        initRecyclerView()
+                    }
+                }
+
+                // TODO я хочу цю функцію винести безпосередньо у клас, а не тримати її у анонімному об'єкті, але тоді нема доступу до onContactSelectedStateChanged(). Як жити?
+                private fun imageViewMyContactsCancelSetOnClickListener() {
+                    binding.imageViewMyContactsCancel?.setOnClickListener {
+                        usersAdapter.unselectItems()
+                        onContactSelectedStateChanged()
+                    }
+                }
+            }
+        )
+    }
+    private var swipeFlags = ItemTouchHelper.END
+    private lateinit var dialogAddContact: AlertDialog
+
+    //region Main code
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initRecyclerView()
+        setSwipeToDelete()
+        createAddContactDialog()
+        handleRecyclerViewContent()
+    }
+
+    override fun setObservers() {
+        parentViewModel.userLiveData.observe(this) { users ->
+            usersAdapter.submitList(users.toMutableList())
+        }
+    }
+
+    override fun setListeners() {
+        imageButtonArrowBackSetOnClickListener()
+        multiSelectedItemsTrashCanOnClickListener()
+    }
+
+    private fun imageButtonArrowBackSetOnClickListener() {
+        binding.imageViewMyContactsArrowBack.setOnClickListener {
+            (parentFragment as ViewPagerFragment).viewPager.currentItem = 0
+        }
+    }
+
+    private fun multiSelectedItemsTrashCanOnClickListener() {
+        binding.buttonRemoveSelectedContacts?.setOnClickListener {
+            removeSelectedItemsFromRecyclerView()
+        }
+    }
+
+    private fun initRecyclerView() {
+        binding.recyclerViewMyContactsContactList.run {
+            adapter = usersAdapter
+            layoutManager = LinearLayoutManager(
+                requireContext(), LinearLayoutManager.VERTICAL, false
+            ) // TODO first parameter context or requireContext() https://stackoverflow.com/a/60402600/20937549
+        }
+    }
+
+    private fun setSwipeToDelete() {
+        val swipeToDeleteCallback = object : SwipeToDeleteCallback() {
+            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
+                val position = viewHolder.adapterPosition
```

Comment:
I suggest you upgrading the recycler view to Sergii`s version but it is not necessary right now